octodns_bind/__init__.py

```diff
@@ -162,12 +163,25 @@ def __init__(
             key_algorithm is not None,
         )
         super().__init__(id)
-        self.host = host
+        self.host = self._host(host)
         self.port = int(port)
         self.key_name = key_name
         self.key_secret = key_secret
         self.key_algorithm = key_algorithm
 
+    def _host(self, host):
+        h = host
+        try:
+            # Determine if IPv4/IPv6 address
+            dns.inet.af_for_address(host)
+        except ValueError:
+            try:
+                h = dns.resolver.resolve(host)[0].address
+            except DNSException as err:
+                raise AxfrSourceZoneTransferFailed(err) from None
```

Comment:
Interesting. Hadn't run across the `from None` thing before.

Reply:
This was the relevant thing I found for it https://stackoverflow.com/questions/52725278/during-handling-of-the-above-exception-another-exception-occurred

In this case the checking of the IP address calls like inet.aton which throws a SyntaxError, which then I think isn’t caught or something, but then the other method throws a ValueError. You end up doing from None to silence the “inner” exceptions (SyntaxError) for the outer one (ValueError)